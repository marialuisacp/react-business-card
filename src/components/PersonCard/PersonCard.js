import React from 'react';
import PropTypes from 'prop-types';

import './PersonCard.scss';
import '../../assets/styles/styles.scss';

const PersonCard = ({ person }) => {
  return (
    <div className='card-person card-info'>
      <div className='circle'></div>
      {person && person.name ?
        <div className='card'>
          <div className='avatar' style={{ backgroundImage: 'url(' + person.image + ')' }}></div>
          <div className='info'>
            <p className='title'>{person.name}</p>
            <p>{person.office}</p>
          </div>
          <div className='add'>
            <p>+</p>
          </div>
        </div>
        : <div className='card outline'></div>}
    </div >
  );
}

PersonCard.propTypes = {
  person: PropTypes.object
};

export default PersonCard;