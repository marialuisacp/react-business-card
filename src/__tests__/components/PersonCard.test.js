import React from 'react';
import renderer from 'react-test-renderer';

import PersonCard from '../../components/PersonCard/PersonCard';

describe('PersonCard', () => {
  it('Component PersonCard should render correctly', () => {
    const props = {
      person: {
        letter: 'D',
        people: [{
          name: 'Danilo Rocha',
          office: 'Desenvolvedor',
          image: './assets/images/people/d01.jpg',
          email: 'danilo.rocha@mundiale.com'
        }]
      }
    }
    const component = renderer.create(
      <PersonCard {...props} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})
