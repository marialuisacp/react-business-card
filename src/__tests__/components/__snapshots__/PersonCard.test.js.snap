// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard Component PersonCard should render correctly 1`] = `
<div
  className="card-person card-info"
>
  <div
    className="circle"
  />
  <div
    className="card outline"
  />
</div>
`;
